transport/socks5/stream_dialer.go

```diff
@@ -23,23 +23,89 @@ import (
 	"github.com/Jigsaw-Code/outline-sdk/transport"
 )
 
+// bufferSize: The maximum buffer size is
+// 3 (1 socks version + 1 method selection + 1 methods)
+// + 1 (auth version) + 1 (username length) + 255 (username) + 1 (password length) + 255 (password)
+const (
```

Comment:
These constants are not helping readability. In fact they are hurting because it makes me go back an forth between the list and the code. Let's keep them inline like they were before. It's easier to follow the protocol that way.